src/Storages/MergeTree/MergeTreeIndexAnnoy.cpp

```diff
@@ -123,6 +123,11 @@ MergeTreeIndexAggregatorAnnoy<Distance>::MergeTreeIndexAggregatorAnnoy(
 template <typename Distance>
 MergeTreeIndexGranulePtr MergeTreeIndexAggregatorAnnoy<Distance>::getGranuleAndReset()
 {
+    /// Seed random generator used for building the trees in Annoy. All other parameters being equal (granularity trees, search_k), this
+    /// theoretically makes search results deterministic as well. In practice, it is hard to achieve that on SQL level because the index
+    /// structure is also affected by implicit things like the row order or how the column data is split into parts.
+    index->set_seed(42);
```

Comment:
(1) I didn't see that any tests were added which check stable results when using Annoy index? If adding seed still doesn't make it possible - why the do it at all?

(2) If setting seed still make sense then it'd be preferable to add a setting for it